cogeo_mosaic/mosaic.py

```diff
@@ -0,0 +1,241 @@
+        Create mosaic definition content.
+
+        Attributes
+        ----------
+        features : List, required
+            List of GeoJSON features.
+        minzoom: int, required
+            Force mosaic min-zoom.
+        maxzoom: int, required
+            Force mosaic max-zoom.
+        quadkey_zoom: int, optional
+            Force mosaic quadkey zoom.
+        accessor: callable, required
+            Function called on each feature to get its identifier (default is feature["properties"]["path"])
+        version: str, optional
+            mosaicJSON definition version (default: 0.0.2).
+        quiet: bool, optional (default: True)
+            Mask processing steps.
+        kwargs: any
+            Options forwarded to MosaicJSON._filter
+
+        Returns
+        -------
+        mosaic_definition : MosaicJSON
+            Mosaic definition.
+
+        """
+        quadkey_zoom = quadkey_zoom or minzoom
+
+        if not quiet:
+            click.echo(f"Get quadkey list for zoom: {quadkey_zoom}", err=True)
+
+        # Find dataset geometries
+        dataset_geoms = polygons(
+            [feat["geometry"]["coordinates"][0] for feat in features]
+        )
+        bounds = list(total_bounds(dataset_geoms))
+
+        tiles = burntiles.burn(features, quadkey_zoom)
+        tiles = [mercantile.Tile(*tile) for tile in tiles]
+
+        mosaic_definition = dict(
+            mosaicjson=version,
+            minzoom=minzoom,
+            maxzoom=maxzoom,
+            quadkey_zoom=quadkey_zoom,
+            bounds=bounds,
+            center=[(bounds[0] + bounds[2]) / 2, (bounds[1] + bounds[3]) / 2, minzoom],
+            tiles={},
+            version="1.0.0",
+        )
+
+        if not quiet:
+            click.echo(f"Feed Quadkey index", err=True)
+
+        # Create tree and find assets that overlap each tile
+        tree = STRtree(dataset_geoms)
+
+        for tile in tiles:
+            quadkey = mercantile.quadkey(tile)
+            tile_geom = polygons(mercantile.feature(tile)["geometry"]["coordinates"][0])
+
+            # Find intersections from rtree
+            intersections_idx = sorted(tree.query(tile_geom, predicate="intersects"))
+            if len(intersections_idx) == 0:
+                continue
+
+            intersect_dataset, intersect_geoms = zip(
+                *[(features[idx], dataset_geoms[idx]) for idx in intersections_idx]
+            )
+
+            dataset = cls._filter(tile, intersect_dataset, intersect_geoms, **kwargs)
+
+            if dataset:
+                mosaic_definition["tiles"][quadkey] = [accessor(f) for f in dataset]
+
+        return cls(**mosaic_definition)
+
+    @classmethod
+    def from_urls(
+        cls,
+        urls: Sequence[str],
+        minzoom: Optional[int] = None,
+        maxzoom: Optional[int] = None,
+        max_threads: int = 20,
+        quiet: bool = True,
+        **kwargs,
+    ):
+        """
+        Create mosaicjson from url of COGs.
```

Comment:
`urls`. Aka we're passing many urls that each point to one COG, not the reverse.